src/crimson/os/seastore/async_cleaner.h

```diff
@@ -294,6 +314,12 @@ class SegmentProvider {
   virtual ~SegmentProvider() {}
 };
 
+class RandomBlockProvider : virtual public Provider {
+public:
+  virtual void update_latest_journal_head_committed(journal_seq_t j) = 0;
```

Comment:
I think it is conceptually same to the existing two interfaces combined:
* allocate_segment()
* update_segment_avail_bytes()

We probably can introduce back the general `update_journal_head()` to maintain a `journal_head` separatly, and for SegmentCleaner, need to validate that it is consistent with the `segments_info_t`.